### ISSUE TITLE
Notification don't work [Update problem still persists for app version: 191.Update it still presists with app version: 192/193 so far.Update problem still persists for version 195,Update issue fixed, tho problem with getting live stream notifications still persists]

### ISSUE THREAD

**Issue by DUBDUBHUB** [futo-org/grayjay-android#238]
Android Device:   **Samsung Galaxy A33 5G**

OS: **Android 13**

GrayJay version: **190**/**191/**192****/**193**/**195**

YouTube plugin version: **139**/**147**/**148**

Odysee plugin version: **30**

**"Feature: Subscription notifications" doesn't work** 

Ok, so I've updated GrayJay to the newest version and the changelog says that we can now get subscription notifications.

 I went to the creators tab and clicked to receive notifications from almost every single person that I am subscribed to and saved the changes that I've made. 
 
![image](https://github.com/futo-org/grayjay-android/assets/150360681/1625165f-4bd6-4629-9e17-695137c5f0a1)

**_I've also given the app permission to send me notifications on my Android phone app settings including pop-ups and stuff_**.

![image](https://github.com/futo-org/grayjay-android/assets/150360681/ad1f22c1-51e6-4437-8c38-6b9da96c418a)
![image](https://github.com/futo-org/grayjay-android/assets/150360681/ea981e7c-bd29-4b00-aef3-617bda8ea7f9)
![image](https://github.com/futo-org/grayjay-android/assets/150360681/457a488d-78f7-4dc1-98c8-8c27e291be5d)

 YouTube has sent me notifications ever since I activated the GrayJay app notification feature, but GrayJay has not.

 I tried to reinstall it, I tried to close YouTube notifications in the app settings to see if that was the problem, but nothing worked.

A friend of mine tried to and I quote:" reset permission, cleared data and cache, install and reinstall with and without backup but to no avail." He runs **Android version: 14** 

 Do I need to change something in my setting or something?

**Comment by Kelvin-FUTO** [comment]
Have you enabled Background Update in settings?
If this is disabled then it will never give you notifications as it isn't checking for them.

With the Background Update you choose the interval you would like it to check.  By default we disable it ("Never") as not everyone their phone randomly doing things in background.

Besides that the behavior is as follows:
Whenever a new video is found during background update (not encountered before), and you have notifications on true for that specific creator, then a notification is created. It does limit this to 5 notifications at a time.

If you have the app open, reloaded your subscriptions, or in any other way already seen that video before, it will not create a notification.

That being said, in the next version (likely 193) I will add a toast message if you enable notifications and have background updating disabled to point this issue out.

**Comment by DUBDUBHUB** [comment]
Yes I did that from the start 
![image](https://github.com/futo-org/grayjay-android/assets/150360681/cb47680a-898b-47fb-8ee2-4d398828c740)
**Should I set them to check every 15 minutes or so perhaps?**
![image](https://github.com/futo-org/grayjay-android/assets/150360681/beaa044e-26b4-49a1-bf84-8b6bd32b1626)

By having the app open I suppose you also mean having it open in the menu section of the phone right? 

Btw thanks for taking the time to respond to my question I appreciate it. 

**Sorry, English isn't my first language by menu section I mean this:**
![image](https://github.com/futo-org/grayjay-android/assets/150360681/18be3f08-cd4a-4917-b4c7-bcf213daaa0f)




It's a bit strange for instance:

I don't shut down my phone at night, and yesterday at like 12.30 pm I received a notification from YouTube of a Livestream made by a creator that I am also subscribed to on GrayJay notifications and everything.

I didn't click on the video or anything I just swiped the notification away closed the phone and that's it.

I still had the 1 hour interval checked and today when I woke up there was no notification, just the ones from Youtube.

I probably had the GrayJay app open in the menu or something 

Btw that sounds like a great feature to have in version 193.

Also I just got a new notification from Youtube while writing this so I will tell you in an hour or so to see if this works GrayJay isn't opened or anything.
 
 
![image](https://github.com/futo-org/grayjay-android/assets/150360681/198bf48e-5c82-42bb-9a89-3b549df2b576)

**Comment by DUBDUBHUB** [comment]
Update: It didn't work as far as I can tell  GrayJay was not open or anything

I got the YouTube notification at 10.00 pm local time where I am from, the background update was set to 1 hour, I waited 1hour and 11 minutes before opening GrayJay 
![image](https://github.com/futo-org/grayjay-android/assets/150360681/1740b8c5-dd5b-4c51-ada4-3206e972074e)
![image](https://github.com/futo-org/grayjay-android/assets/150360681/3d37a2cb-fec7-4d57-9252-1d8b34c82566)
![image](https://github.com/futo-org/grayjay-android/assets/150360681/83cbc203-43bc-4163-913d-52d808fce657)

The next notification I received  was at 11.15pm local time, I set the background update to 15 minutes I waited 20 minutes before opening GrayJay
 
![image](https://github.com/futo-org/grayjay-android/assets/150360681/7c2fa8dd-f972-4d03-8c61-8cd845a330ce)
![image](https://github.com/futo-org/grayjay-android/assets/150360681/bab13af0-8b69-46c3-804d-4f7cb71e3b15)
![image](https://github.com/futo-org/grayjay-android/assets/150360681/d1cf7a98-2dee-4152-8324-cbce5fad2e33)
![image](https://github.com/futo-org/grayjay-android/assets/150360681/9cda0176-af97-4fad-94ca-0aa729bcd106)

I am going to do another test with the following setting activated to see if that solves the issue since it wasn't activated the previous times 

 
![image](https://github.com/futo-org/grayjay-android/assets/150360681/06bc8856-97bb-4586-988b-883223b77d78)

**Comment by DUBDUBHUB** [comment]
**I updated the previous comment with extra information please check**

**Comment by Kelvin-FUTO** [comment]
@DUBDUBHUB Perhaps something is still going wrong.
Note, Youtube notifications are unrelated to Grayjay.

One thing you could test to make sure your background updater is working at all is the following:
 - Go to settings
 - Scroll all the way to the bottom
 - Press the version code (eg. 193) 5 times
 - Developer settings should now be available at the bottom
 - Click the purple button at the bottom
 - Tap "Test Background Worker" once.
 If things go as they should, you should see an ongoing notification of your subscriptions being updated.
 If this is NOT the case, it means something is either blocking notifications, or there is nothing to update.

 I'll investigate it some more on Monday.

**Comment by DUBDUBHUB** [comment]
Hey Kelvin thanks again for taking the time to respond I appreciate it 
This notification appears and then it stops abruptly at like 47,48,49 and 58 usually is this what you mean by"If this is NOT the case, it means something is either blocking notifications, or there is nothing to update."?
![image](https://github.com/futo-org/grayjay-android/assets/150360681/088d4e24-1014-46bd-a895-e594e8be3a11)

**Here are the settings in the developer sections:**
![image](https://github.com/futo-org/grayjay-android/assets/150360681/ac92f0bd-848c-480b-b12f-e1b65b47445a)

So what can I do about it, so that it no longer blocks notifications?  Of course, If it isn't too much to ask.

Also thanks for continuing to investigate it

**Comment by DUBDUBHUB** [comment]
Here's what I could find so far regarding notifications 
On "Choose apps that can show content in Lock screen notifications" GrayJay is allowed
![image](https://github.com/futo-org/grayjay-android/assets/150360681/2827bb3e-447d-4f48-9dd8-2793d535a1f5)
**On "Apps to show as brief" same thing** 
![image](https://github.com/futo-org/grayjay-android/assets/150360681/9dda37ce-ec63-4221-9ed3-98b5ce7a391c)
**On "Choose which apps are allowed to send you notifications" same thing as well** 
![image](https://github.com/futo-org/grayjay-android/assets/150360681/b27eb188-a257-4d67-ac38-eb3ffdfbbf73)
**These are my "Advanced settings"**
![image](https://github.com/futo-org/grayjay-android/assets/150360681/4ae15fa7-fceb-43ce-b474-09d3e4a0bc2c)
**And here is my notification category** 
![image](https://github.com/futo-org/grayjay-android/assets/150360681/1b796a96-58bc-4cc0-b401-fd00f2518ba9)

**Comment by DUBDUBHUB** [comment]
Here are other settings regarding notifications as well 
![image](https://github.com/futo-org/grayjay-android/assets/150360681/4dcc28d7-96bc-4dcd-b80c-1c9b305cfb16)
![image](https://github.com/futo-org/grayjay-android/assets/150360681/f02e182e-792e-43b3-9829-5a1673bd50cf)
![image](https://github.com/futo-org/grayjay-android/assets/150360681/ac9cb2b8-16fb-4126-a254-02d97fcb6d8b)

**Comment by DUBDUBHUB** [comment]
Update this is what my friend sent me:
" did tap the "Test Background Worker" and I got the ongoing notification but still the notifications in the grayjay app not working at all not even for the Twitch plugin."

"Yes, it stops and didn't finish like it disappears after 1 sec or less."
![image](https://github.com/futo-org/grayjay-android/assets/150360681/63f59ec3-76ed-4eef-ae90-463fa2b67075)
![image](https://github.com/futo-org/grayjay-android/assets/150360681/fdeedf3f-dcc2-467d-a9f5-51527d0044b0)

**Comment by DUBDUBHUB** [comment]
Here are some other updates regarding notifications 
it stopped at 86 out of 88
![image](https://github.com/futo-org/grayjay-android/assets/150360681/1e813cb0-cc10-4501-a460-8ad3a9b8df0c)
![image](https://github.com/futo-org/grayjay-android/assets/150360681/960c73bb-427c-47b4-81e7-f2ea3c9bd953)
![image](https://github.com/futo-org/grayjay-android/assets/150360681/9f1bcab0-7c3c-49c7-b2a0-c56e360b7647)

**Comment by DUBDUBHUB** [comment]
Hey @Kelvin-FUTO  sorry to bother you I know you are busy man solving more important issues right now but do you have any news regarding this problem?

**Comment by Kelvin-FUTO** [comment]
@DUBDUBHUB It should be solved in upcoming 194, should be released tomorrow I think. 
It ended up being something fairly obvious and thus I think this is it.
I tested it and it appears to work at a glance, but do let me know after 194 released if it ended up working. (The same requirements from before apply, aka unseen content in-app)

**Comment by DUBDUBHUB** [comment]
Ok thanks man, I appreciate it a lot. I will let you know once I update it to 194. Have a great day.

**Comment by DUBDUBHUB** [comment]
Hey @Kelvin-FUTO  the problem still persists. I received 2 notifications from YouTube (which I use to check if I get notifications at all) at 10:30 pm local, The background update is still set to 15 minutes I waited 20 minutes GrayJay hasn't sent any notification.
GrayJay was not open or anything. I haven't changed any settings or anything notifications are still allowed and all the other pictures with the settings. The continuous notification when activating background work testing still stops abruptly at random numbers such as 50 and 68.
 
![image](https://github.com/futo-org/grayjay-android/assets/150360681/aec69381-c758-41e3-a888-7ddb4fd0f44e)
![image](https://github.com/futo-org/grayjay-android/assets/150360681/ec32cdee-cb10-465a-9a7d-221dc1edc0f1)
![image](https://github.com/futo-org/grayjay-android/assets/150360681/e181a20a-8627-4906-a3bf-44657644316a)
![image](https://github.com/futo-org/grayjay-android/assets/150360681/98b81b81-fa2e-48c4-96a4-d852a2320054)
![image](https://github.com/futo-org/grayjay-android/assets/150360681/071499a3-879a-4d6d-ae34-662ed0c96ee0)
![image](https://github.com/futo-org/grayjay-android/assets/150360681/6d644312-61a4-42bc-977e-5714025a52b6)
![image](https://github.com/futo-org/grayjay-android/assets/150360681/b4f5acbf-eb06-4c20-beec-25509586ae7e)
![image](https://github.com/futo-org/grayjay-android/assets/150360681/0130d63e-c382-4665-8a11-1abb9e1bdde4)

**Comment by DUBDUBHUB** [comment]
Update they work for my friend at least the one that has Android 14 
![image](https://github.com/futo-org/grayjay-android/assets/150360681/68258e75-1d11-4d43-a79c-c0b56fb1d1b0)
![image](https://github.com/futo-org/grayjay-android/assets/150360681/c3d71059-94df-4d38-876a-b57f45149e10)

**Comment by DUBDUBHUB** [comment]
Update from my friend:"Nothing for today, uninstall and reinstall still nothing, btw I have twitch plugins and notification not working as well that means it's general issue."

**Comment by Kelvin-FUTO** [comment]
To cover all previously mentioned stuff:

Reinstalling is not going to change anything, and might actually work against you as it resets various stuff.

I've seen notifications work as expected (as long as you have it enabled for a channel).
How often it is checked is partially up to the phone. We have no control over when your phone allows us to check.
Especially if you have a Chinese brand or a Samsung, they heavily try to avoid any battery usage.
When we say the update interval is 15 minutes, it means we tell Android to let us update every 15 minutes. It does not mean it actually happens sadly, as some brands simply refuse to do it, or do it whenever it is convenient (for battery).
The following website may offer some solutions depending on your brand: https://dontkillmyapp.com/

The Youtube plugin version is not the same as the app version. They are fundamentally distinct and have their own versioning. If you have installed the apk from our website it will automatically be updated to the latest whenever we update the app though.

Comparisons to Youtube notifications aren't very comparable since Youtube most likely uses push notifications from their servers, not from their app.

What isn't implemented yet but is planned is make scheduled notifications work, meaning that if a livestream or video is planned (publicly), Grayjay can schedule a notification in the future which should appear on time.
But this only works for some Youtube livestreamers as far as I know Twitch doesn't have such a feature.

I have not explicitly tested Twitch notifications, but I've enabled a few on my phone to see if they do work on my phone.

The cut off at 50-60 might just be because there is no point in updating some subscriptions, or they are only updated once per day (eg. If a channel doesn't update for multiple months, we only check daily).
I don't know how many subscriptions you have, or what kind of subscriptions you have, so I can't determine if 50-60 is an accurate number. The same applies to streams, if you have streams fetching enabled for a channel, and that channel never streams, it might ignore that as well thus skipping it. I will likely change this so that cached calls are not shown in the progress bar as it appears to be confusing for some.

For reference, I have about 100 subscriptions, of which 20-ish are not regularly updated as they are channels that infrequently upload.

**Comment by DUBDUBHUB** [comment]
Hey @Kelvin-FUTO thank you so much for your suggestion in regards to the website it works now, at least for me.
It updates every 15 minutes approximately as you said.
![image](https://github.com/futo-org/grayjay-android/assets/150360681/ee0739a5-4b9d-4b11-9bb2-fe81c759a36e)
There are still some problems with getting short content format like videos that are less than a minute long and livestream notifications other than that it works pretty well so far.

**Comment by DUBDUBHUB** [comment]
For instance, this is what my friend said  before he tried the website that you suggested but I have similar issues with YouTube plugin livestreams on Grayjay so yea just wanted to let you know what we encountered so far:

"This is what I got today Two notifications at 10:16 AM but if you check the third picture you'll find there are three videos has been uploaded in that time period (two of them was 11 hours ago and the third one is 15 hours ago) two of them from YouTube and the third one from Twitch.

Long story short, I should get 3 notifications but I got 2 only,..."
![image](https://github.com/futo-org/grayjay-android/assets/150360681/829303f7-2654-46da-8d5f-d549cd037904)
and
**"I got another one Btw, there is a plugin update. I update the plugin but no notifications after the update."**
![image](https://github.com/futo-org/grayjay-android/assets/150360681/59b3acf4-c438-4433-9ad5-ce53dd144454)
